calendar/engine/user_test.go

```diff
@@ -0,0 +1,700 @@
+	}{
+		{
+			name:    "error expanding mattermost user",
+			eventID: "testEventID",
+			setupMock: func() {
+				user.User = nil
+				user.MattermostUser = nil
+				mockPluginAPI.EXPECT().GetMattermostUser("testMMUserID").Return(nil, errors.New("some error occurred while getting mattermost user"))
+			},
+			assertions: func(t *testing.T, err error) {
+				require.Error(t, err)
+				require.EqualError(t, err, "some error occurred while getting mattermost user")
+			},
+		},
+		{
+			name:    "success expanding mattermost user",
+			eventID: "testEventID",
+			setupMock: func() {
+				user.User = nil
+				user.MattermostUser = nil
+				mockPluginAPI.EXPECT().GetMattermostUser("testMMUserID").Return(&model.User{}, nil)
+			},
+			assertions: func(t *testing.T, err error) {
+				require.NoError(t, err)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.setupMock()
+
+			err := mscalendar.ExpandMattermostUser(user)
+
+			tt.assertions(t, err)
+		})
+	}
+}
+
+func TestGetTimezone(t *testing.T) {
+	mscalendar, mockStore, _, _, _, mockClient, _ := MockSetup(t)
+
+	user := &User{
+		User:             &store.User{Settings: store.Settings{}, Remote: &remote.User{ID: "testRemoteID"}},
+		MattermostUserID: "testMMUserID",
+	}
+
+	tests := []struct {
+		name       string
+		eventID    string
+		setupMock  func()
+		assertions func(t *testing.T, err error)
+	}{
+		{
+			name:    "error loading remote user",
+			eventID: "testEventID",
+			setupMock: func() {
+				user.User = nil
+				mockStore.EXPECT().LoadUser("testMMUserID").Return(nil, errors.New("error filtering user")).Times(1)
+			},
+			assertions: func(t *testing.T, err error) {
+				require.Error(t, err)
+				require.EqualError(t, err, "It looks like your Mattermost account is not connected to testDisplayName. Please connect your account using `/testCommandTrigger connect`.: error filtering user")
+			},
+		},
+		{
+			name:    "error getting mailbox setting",
+			eventID: "testEventID",
+			setupMock: func() {
+				user.User = &store.User{Settings: store.Settings{}, Remote: &remote.User{ID: "testRemoteID"}}
+				mockClient.EXPECT().GetMailboxSettings("testRemoteID").Return(nil, errors.New("error occurred getting mailbox settings"))
+			},
+			assertions: func(t *testing.T, err error) {
+				require.Error(t, err)
+				require.EqualError(t, err, "error occurred getting mailbox settings")
+			},
+		},
+		{
+			name:    "success getting mailbox setting",
+			eventID: "testEventID",
+			setupMock: func() {
+				user.User = &store.User{Settings: store.Settings{}, Remote: &remote.User{ID: "testRemoteID"}}
+				mockClient.EXPECT().GetMailboxSettings("testRemoteID").Return(&remote.MailboxSettings{TimeZone: "mockTimeZone"}, nil)
+			},
+			assertions: func(t *testing.T, err error) {
+				require.NoError(t, err)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.setupMock()
+
+			_, err := mscalendar.GetTimezone(user)
```

Comment:
Should we test the first return vallue?

Reply:
the first value is returned by us only from the mocked function, so no point testing it
`mockClient.EXPECT().GetMailboxSettings("testRemoteID").Return(&remote.MailboxSettings{TimeZone: "mockTimeZone"}, nil)`